### PR TITLE
refactor: move out status management into StatusManager

### PR DESCRIPTION
**Pull request by Alex-Izquierdo**
Refactor the activation manager to move out the status management into a parent class "StatusManager". This is used when we want to change the status of activations in the orchestrator prior to the container livecycle. 

Jira: https://issues.redhat.com/browse/AAP-22452

### CODE REVIEW

File: src/aap_eda/services/activation/manager.py [hunk 1]
```diff
@@ -76,7 +76,58 @@ def _run_with_lock(self: HasDbInstance, *args, **kwargs):
     return _run_with_lock
 
 
-class ActivationManager:
+class StatusManager:
```
Comment (mkanoor) [inline src/aap_eda/services/activation/manager.py:79]: @Alex-Izquierdo Can we move this to a separate file? This file is already too big. It also allows us to carve out other pieces into smaller classes like the Restart Policies

Comment (Alex-Izquierdo) [reply src/aap_eda/services/activation/manager.py]: Hi @mkanoor I've created status_manager file and renamed manager to activation_manager. Let me know if that's fine to you.

**Review by hsong-rh — APPROVED** [review]

File: src/aap_eda/services/activation/status_manager.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+from aap_eda.core.enums import ActivationStatus, ProcessParentType
+
+LOGGER = logging.getLogger(__name__)
+
+
+class HasDbInstance(tp.Protocol):
+    db_instance: tp.Any
+
+
+def run_with_lock(func: tp.Callable) -> tp.Callable:
```
Comment (mkanoor) [inline src/aap_eda/services/activation/status_manager.py:32]: @Alex-Izquierdo Is this generic enough to be moved into a util like db_util and shared from there for everyone to use.

Comment (Alex-Izquierdo) [reply src/aap_eda/services/activation/status_manager.py:32]: I don't think so, at least for now. It's bound to a "has_db_instance" type which currently is only the activation_manager. In the future we might want to change, but for now I would say it is strictly related with the activation manager.

Comment (Alex-Izquierdo) [reply src/aap_eda/services/activation/status_manager.py:32]: Also, This refactor is intended to be small, only to split the status manager which I need it to be able to update the activation from the orchestrator. I would prefer to consider any other changes in different PR's.

**Review by mkanoor — APPROVED** [review]

**Pull request merged**